sound/soc/sof/topology.c

```diff
@@ -492,6 +502,17 @@ static int get_token_u16(void *elem, void *object, u32 offset, u32 size)
 	return 0;
 }
 
+static int get_token_uuid(void *elem, void *object, u32 offset, u32 size)
+{
+	struct snd_soc_tplg_vendor_uuid_elem *velem = elem;
+	u32 *val = (u32 *)((u8 *)object + offset);
+	guid_t uuid;
+
+	import_guid(&uuid, velem->uuid);
+	*val = sof_get_uid_entry(&uuid);
```

Comment:
@mrajwa I'd still want to double check do we really need to use the dictionary here, or could we pass the full UUID in the IPC messages. I understand for tracing, you do not want encode the long UUID for each entry, so a dictionary is warranted, but for IPCs such as COMP_NEW, the overhead does not seem much.

Reply:
I second this. Also, if I understand correctly, not all components will have a UUID isnt it? The generic ones such as host,buffer, dai etc dont need one right?

Reply:
> @mrajwa I'd still want to double check do we really need to use the dictionary here, or could we pass the full UUID in the IPC messages. I understand for tracing, you do not want encode the long UUID for each entry, so a dictionary is warranted, but for IPCs such as COMP_NEW, the overhead does not seem much.

Should be @mmaka1 .

Reply:
> I second this. Also, if I understand correctly, not all components will have a UUID isnt it? The generic ones such as host,buffer, dai etc dont need one right?

I think in the FW, we are adding UUID for all components, including host/buffer/dai/volume.

I am not enabling non-process/effect ones in this series, but keep the possibility for future.

Reply:
The UUID in firmware is currently stored in a discarded section, which means firmware doesn't actually have access to the UUID data itself. And on APL at least I don't think we should move the UUID itself in firmware (increasing sizes from 4 bytes to 32 bytes per component)

Reply:
> The UUID in firmware is currently stored in a discarded section, which means firmware doesn't actually have access to the UUID data itself. And on APL at least I don't think we should move the UUID itself in firmware (increasing sizes from 4 bytes to 32 bytes per component)

@paulstelian97 This is changed in the latest design, UUID will be saved to the SRAM rodata section, please see it here: thesofproject/sof-docs#245